services/beaconchain/src/test/java/tech/pegasys/teku/services/beaconchain/WeakSubjectivityInitializerTest.java

```diff
@@ -222,9 +222,11 @@ public void validateInitialAnchor_atMostRecentPotentiallyFinalizedEpoch() {
     final UInt64 currentSlot = compute_start_slot_at_epoch(currentEpoch);
 
     final DataStructureUtil dataStructureUtil = new DataStructureUtil();
-    final AnchorPoint anchor = dataStructureUtil.randomAnchorPoint(currentEpoch.minus(2));
+    final AnchorPoint anchor =
+        dataStructureUtil.randomAnchorPoint(
+            currentEpoch, spec.getForkSchedule().getFork(currentEpoch));
```

Comment:
The original version of this test was using an anchor point from `currentEpoch.minus(2)` which has been lost with this change and that's why it's failing - we can't have finalized epoch 10 if we're only up to the first slot in that epoch.

We can fix the deprecation warnings in this test and the failure by changing it to:

```
    
  @Test
  public void validateInitialAnchor_atMostRecentPotentiallyFinalizedEpoch() {
    final UInt64 currentEpoch = UInt64.valueOf(10);
    final UInt64 currentSlot = spec.computeStartSlotAtEpoch(currentEpoch);

    final DataStructureUtil dataStructureUtil = new DataStructureUtil(spec);
    final UInt64 anchorEpoch = currentEpoch.minus(2);
    final AnchorPoint anchor =
        dataStructureUtil.randomAnchorPoint(
            anchorEpoch, spec.getForkSchedule().getFork(anchorEpoch));

    // Should not throw
    initializer.validateInitialAnchor(anchor, currentSlot, spec);
  }
```